core/sql/executor/ExHdfsScan.cpp

```diff
@@ -410,16 +411,81 @@ ExWorkProcRetcode ExHdfsScanTcb::work()
 
 	    hdfsScanBufMaxSize_ = hdfsScanTdb().hdfsBufSize_;
 
+            dataModCheckDone_ = FALSE;
+
 	    if (numRanges_ > 0)
-              step_ = INIT_HDFS_CURSOR;
+              step_ = CHECK_FOR_DATA_MOD;
             else
-              step_ = DONE;
+              step_ = CHECK_FOR_DATA_MOD_AND_DONE;
 	  }
 	  break;
 
+        case CHECK_FOR_DATA_MOD:
+        case CHECK_FOR_DATA_MOD_AND_DONE:
+          {
+            char * dirPath = hdfsScanTdb().hdfsRootDir_;
```

Comment:
At this time, data modification check for partitioned tables is not being
supported. Once we support partitioned tables, we will need to enhance
code to handle them.

Here is the code in GenRelScan.cpp disabling it:

  // partitioned files. Skip data mod check for them.
 +      if (numOfPartLevels > 0)
 +        hdfsRootDir = NULL;